build/tools/Automation/DotNet/DotNet.cs

```diff
@@ -0,0 +1,236 @@
+                    "platforms/dotnet/versions/bullseye/versionsToBuild.txt",
+                    "platforms/dotnet/versions/buster/versionsToBuild.txt",
+                    "platforms/dotnet/versions/focal-scm/versionsToBuild.txt",
+                    "platforms/dotnet/versions/stretch/versionsToBuild.txt",
+            };
+            foreach (string versionsToBuildTxtFile in versionsToBuildTxtFiles)
+            {
+                string line = $"\n{platformConstant.Version}, {platformConstant.Sha},";
+                File.AppendAllText(versionsToBuildTxtFile, line);
+
+                // sort
+                Console.WriteLine($"Updating {versionsToBuildTxtFile}...");
+                var contents = File.ReadAllLines(versionsToBuildTxtFile);
+                Array.Sort(contents);
+                File.WriteAllLines(versionsToBuildTxtFile, contents);
+            }
+        }
+
+        private static Dictionary<string, Constant> GetYamlDotNetConstants(List<Constant> yamlContents)
+        {
+            Dictionary<string, Constant> dotNetConstants = new Dictionary<string, Constant>();
+            foreach (var constant in yamlContents)
+            {
+                if (constant.Name == Constants.DotNetSdkKey ||
+                    constant.Name == Constants.DotNetRuntimeKey)
+                {
+                    dotNetConstants.Add(constant.Name, constant);
+                }
+            }
+            return dotNetConstants;
+        }
+
+        private static string GenerateDotNetConstantKey(PlatformConstant platformConstant)
+        {
+            string[] splitVersion = platformConstant.Version.Split('.');
+            string majorVersion = splitVersion[0];
+            string minorVersion = splitVersion[1];
+            string majorMinor = majorVersion + minorVersion;
+            string constant;
+            if (platformConstant.VersionType.Equals(Constants.SdkName))
+            {
+                // TODO: add try catch in case the integer is un-parseable.
+                int majorVersionInt = int.Parse(majorVersion);
+
+                // dotnet/dotnetcore are used based on the major version
+                string prefix = majorVersionInt < 5 ? $"dot-net-core" : "dot-net";
+                constant = $"{prefix}-{majorMinor}-sdk-version";
+            }
+            else
+            {
+                constant = $"{platformConstant.VersionType}-app-{majorMinor}";
+            }
+
+            // TODO: add Logger.Debug the constant that is generated
+            return constant;
+        }
+
+        private static string GetSha(List<FileObj> files)
+        {
+            HashSet<string> tarFileNames = new HashSet<string>() {
+                "dotnet-sdk-linux-x64.tar.gz",
+                "dotnet-runtime-linux-x64.tar.gz",
+                "aspnetcore-runtime-linux-x64.tar.gz",
+            };
+            foreach (var file in files)
+            {
+                if (tarFileNames.Contains(file.Name))
```

Comment:
nit: I think it would be easier if we check that the file matched the pattern `*-linux-x64.tar.gz` using RegEx (expression would need to be updated, but you get the idea)

Reply:
Updated with regex.